src/main/java/org/jitsi/jibri/capture/ffmpeg/executor/impl/MacFfmpegExecutor.kt

```diff
@@ -0,0 +1,64 @@
+package org.jitsi.jibri.capture.ffmpeg.executor.impl
+
+import org.jitsi.jibri.capture.ffmpeg.executor.FfmpegExecutor
+import org.jitsi.jibri.capture.ffmpeg.executor.FfmpegExecutorParams
+import org.jitsi.jibri.sink.Sink
+import org.jitsi.jibri.util.debug
+import java.io.File
+import java.util.concurrent.TimeUnit
+import java.util.logging.Logger
+
+/**
+ * Ffmpeg execution specific to Mac OS
+ */
+class MacFfmpegExecutor : FfmpegExecutor
+{
+    private val logger = Logger.getLogger(this::class.simpleName)
+    var currentFfmpegProc: Process? = null
+
+    override fun launchFfmpeg(ffmpegExecutorParams: FfmpegExecutorParams, sink: Sink)
+    {
+        val ffmpegCommandMac = "" +
+                "ffmpeg -y -v info " +
+                "-thread_queue_size ${ffmpegExecutorParams.queueSize} " +
+                "-f avfoundation " +
+                "-framerate ${ffmpegExecutorParams.framerate} " +
+                "-video_size ${ffmpegExecutorParams.resolution} " +
+                "-i ${ffmpegExecutorParams.videoInputDevice}:${ffmpegExecutorParams.audioInputDevice} " +
+                "-vsync 2 " +
+                "-acodec aac -strict -2 -ar 44100 " +
+                "-c:v libx264 -preset ${ffmpegExecutorParams.videoEncodePreset} " +
+                "${sink.getOptions()} -pix_fmt yuv420p -crf ${ffmpegExecutorParams.h264ConstantRateFactor} " +
+                "-g ${ffmpegExecutorParams.gopSize} -tune zerolatency " +
+                "-f ${sink.getFormat()} ${sink.getPath()}"
+
+        val pb = ProcessBuilder(ffmpegCommandMac.split(" "))
+        pb.redirectOutput(File("/tmp/ffmpeg.out"))
+        pb.redirectError(File("/tmp/ffmpeg.out"))
+
+        logger.info("running ffmpeg command:\n $ffmpegCommandMac")
+        //currentFfmpegProc = Runtime.getRuntime().exec(ffmpegCommandMac)
+        currentFfmpegProc = pb.start()
+        logger.debug("launched ffmpeg, is it alive? ${currentFfmpegProc?.isAlive()}")
+    }
+
+    override fun isAlive(): Boolean = currentFfmpegProc?.isAlive ?: false
+
+    override fun getExitCode(): Int? = currentFfmpegProc?.exitValue()
+
+    override fun stopFfmpeg() {
+        currentFfmpegProc?.pid()?.let {
+            logger.info("sending SIGINT to ffmpeg proc ${currentFfmpegProc?.pid()}")
```

Comment:
inside of the let you can just refer to `it` and not have to do any null checks

Reply:
ah much cleaner.  thanks.